### PR TITLE
Video post player[].embed_code from string or bool

### LINKED ISSUE #6: json: cannot unmarshal bool into Go struct field .embed_code of type string
https://github.com/tumblr/tumblr.go/blob/01c315a92daab3e8af9df49da1363cc60421fa70/post.go#L181 

I guess it should be a bool instead of a string ?

```
json: cannot unmarshal bool into Go struct field .embed_code of type string
```

### PR DESCRIPTION
**Pull request by jasonpenny**
This fixes #6, video post player[].embed_code can be a string or boolean, this defines a type `StringOrBool` with a custom UnmarshalJSON function.

I also ran this file through `gofmt`.

This test code demonstrates the broken behavior and verifies this fix.

```go
package main

import (
	"log"
	"net/url"
	"os"

	tumblr "github.com/tumblr/tumblr.go"
	tumblr_go "github.com/tumblr/tumblrclient.go"
)

func main() {
	consumer_key := os.Getenv("TUMBLR_CONSUMER_KEY")
	consumer_secret := os.Getenv("TUMBLR_CONSUMER_SECRET")
	token := os.Getenv("TUMBLR_TOKEN")
	token_secret := os.Getenv("TUMBLR_TOKEN_SECRET")

	client := tumblr_go.NewClientWithToken(consumer_key, consumer_secret, token, token_secret)

	get(client, "animetube", "163255975138")
	get(client, "animetube", "163255974438")
}

func get(client *tumblr_go.Client, blogname, post_id string) {
	params := url.Values{}
	params.Set("id", post_id)
	resp, err := tumblr.GetPosts(client, blogname, params)
	if err != nil {
		log.Fatal(err)
	}

	allPosts, err := resp.All()
	if err != nil {
		log.Fatal(err)
	}

	for _, p := range allPosts {
		switch pt := p.(type) {
		case *tumblr.VideoPost:
			for _, y := range pt.Players {
				log.Print(y)
			}
		}
	}
	log.Println("---")
}
```

@ceyko

### CODE REVIEW

File: post.go [hunk 1]
```diff
@@ -4,17 +4,17 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"reflect"
 	"net/url"
+	"reflect"
 )
 
 // Representation of a list of Posts
 type Posts struct {
```
Comment (byxorna) [inline post.go]: The comments here dont align with https://blog.golang.org/godoc-documenting-go-code. This isnt introduced in this PR, but would be nice to true up while we are in here :)

File: post.go [hunk 2]
```diff
@@ -195,22 +195,44 @@ type VideoPost struct {
...
+	Width  uint32 `json:"width"`
+	Url    string `json:"url"`
+}
+
+// StringOrBool is a string that can be unmarshalled from a string or a boolean value.
+type StringOrBool string
+
+// UnmarshalJSON implements the json.Unmarshaler interface to ingest strings or boolean values.
+func (sb *StringOrBool) UnmarshalJSON(b []byte) error {
+	if b[0] == '"' {
```
Comment (byxorna) [inline post.go:221]: id use string literals with backticks here instead, to avoid confusing quotation

**Review by byxorna — APPROVED** [review]
this could use some CI hooked up to auto fmt/lint, but its outside of the scope of this jawn :)

Nice work!

File: post.go (hunk 2 above)
Comment (jasonpenny) [reply post.go:221]: This would then need to be
```go
if b[0] == `"`[0] {
```
which looks worse, I think

**Pull request merged**